src/components/ec-currency-input/ec-currency-input.vue

```diff
@@ -110,166 +110,152 @@
+    type: String,
+  },
+  isWarning: {
+    type: Boolean,
+    default: false,
+  },
+  errorMessage: {
+    type: String,
+  },
+  errorTooltipMessage: {
+    type: String,
+  },
+  currencies: {
+    type: Array,
+    default: () => [],
   },
-  computed: {
-    id() {
-      return `ec-currency-input-field-${this.uid}`;
-    },
-    errorId() {
-      return this.isInvalid ? `ec-currency-input-field-${this.uid}` : null;
-    },
-    isInvalid() {
-      return !!this.errorMessage;
-    },
-    currenciesItems() {
-      return this.currencies.map(currency => ({ text: currency, value: currency, id: currency }));
-    },
-    currenciesTooltipOptions() {
-      const { content, placement = 'top' } = this.disabledCurrenciesTooltip || {};
-      if (!content) {
-        return null;
-      }
-      return { content, placement };
-    },
-    currencyModel: {
-      get() {
-        return this.currenciesItems.find(item => item.value === this.modelValue.currency);
-      },
-      set(item) {
-        this.$emit('update:modelValue', { ...this.modelValue, currency: item.value });
-      },
-    },
-    amountModel: {
-      get() {
-        return this.modelValue.amount;
-      },
-      set(value) {
-        this.$emit('update:modelValue', { ...this.modelValue, amount: value });
-      },
-    },
+  currenciesAreLoading: {
+    type: Boolean,
+    default: false,
   },
-  methods: {
-    getPopoverStyle() {
-      if (this.$refs.popoverWidthReference) {
-        return {
-          width: `${this.$refs.popoverWidthReference.offsetWidth}px`,
-        };
-      }
-
-      return null;
-    },
-    onAmountChange(evt) {
-      this.$emit('change', evt);
-      this.$emit('amount-change', evt);
-    },
-    onCurrencyChange(evt) {
-      this.currenciesHasFocus = true;
-      this.$emit('change', evt);
-      this.$emit('currency-change', evt);
-    },
-    onFocusCurrency() {
-      this.currenciesHasFocus = true;
-      this.$emit('focus');
-    },
-    prefixedDataTest(dataTestSuffix) {
-      const dataTestPrefix = this.$attrs['data-test'];
-      if (dataTestPrefix) {
-        const dataTestPrefixes = [...dataTestPrefix.split(' '), 'ec-currency-input'];
-        return dataTestPrefixes.map(dataTest => `${dataTest}__${dataTestSuffix}`).join(' ');
-      }
-
-      return `ec-currency-input__${dataTestSuffix}`;
-    },
+  disabledCurrenciesTooltip: {
+    type: Object,
   },
+  isCurrenciesDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isAmountDisabled: {
+    type: Boolean,
+    default: false,
+  },
+  isSensitive: {
+    type: Boolean,
+    default: false,
+  },
+  amountPlaceholder: {
+    type: String,
+  },
+  searchCurrencyPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+  noCurrenciesText: {
+    type: String,
+    default: 'No results found',
+  },
+});
+
+const emit = defineEmits(['update:modelValue', 'change', 'focus', 'amount-change', 'currency-change', 'open', 'after-open']);
+
+const uid = getUid();
```

Comment:
This component is very similar to the phone-number-input refactored in the other PR. I commented there that we should be grouping chunks of code together, e.g. like this: https://github.com/Ebury/chameleon/pull/1293#discussion_r980251636. 

The options API was grouping state together, computed props into one place, methods into one place, but the functionality related to them was scattered across the whole component. `<script setup>` allows us to move state, computed props and methods that are related together. We can use the same grouping as in ec-phone-number-input.

Reply:
sure, done